week2/index.js

```diff
@@ -0,0 +1,128 @@
+"use strict";
+
+const fs = require('fs');
+
+const TODOS_FILENAME = process.env["TODOS_FILENAME"] || "to-do-list.json";
+
+let myTodoList = []
+
+
+// Helper functions definitions
+
+let readFileFunction = (filename) => {
+    return new Promise((resolve, reject) => {
+        fs.readFile(filename, "utf8", (err, data) => {
+            if (err) {
+                if (err.code === "ENOENT") {
+                    // No such file or directory
+                    console.log("I'm sorry, friend, but that file doesn't exist.");
+                } else if (err.code === "EACCES") {
+                    console.log("That file exists but you don't have permission.");
+                } else {
+                    console.log(`Uh oh, an unknown error occured: ${err.message}`);
+                }
+                process.exit()
+            }
+            //  console.log(`Successfully read the file: ${filename}`)
+            resolve(data)
+        })
+    })
+};
+
+let writeFileFunction = (filename, dataAsString) => {
+    return new Promise((resolve, reject) => {
+        fs.writeFile(filename, dataAsString, (err) => {
+            if (err) reject(console.log(`error happened while writing the file \n${err.message}`));
```

Comment:
In this case, your reject is called with `undefined`.

Why? Because the return value of the parameter it's being passed (console.log) is undefined - console.log prints but doesn't return anything.

You probably want to `reject("Error message")`, and then have the caller console.log it in their `.catch()`.